scipy/stats/tests/test_discrete_basic.py

```diff
@@ -14,31 +14,22 @@
 vals = ([1, 2, 3, 4], [0.1, 0.2, 0.3, 0.4])
 distdiscrete += [[stats.rv_discrete(values=vals), ()]]
 
+# For these distributions, test_discrete_basic only runs with test mode full
+distslow = {'zipfian', 'nhypergeom'}
+
 
 def cases_test_discrete_basic():
     seen = set()
     for distname, arg in distdiscrete:
-        yield distname, arg, distname not in seen
+        if distname in distslow:
+            yield pytest.param(distname, arg, distname, marks=pytest.mark.slow)
```

Comment:
I was wondering how the tested get marked for pytest to still pickup the slow marker, it's in this line. LGTM.